ts/components/conversation/media-gallery/groupMessagesByDate.ts

```diff
@@ -0,0 +1,164 @@
+/**
+ * @prettier
+ */
+import moment from 'moment';
+import { compact, groupBy, sortBy } from 'lodash';
+
+import { Message } from './propTypes/Message';
+// import { missingCaseError } from '../../../util/missingCaseError';
+
+type StaticSectionType = 'today' | 'yesterday' | 'thisWeek' | 'thisMonth';
+type YearMonthSectionType = 'yearMonth';
+
+interface GenericSection<T> {
+  type: T;
+  messages: Array<Message>;
+}
+type StaticSection = GenericSection<StaticSectionType>;
+type YearMonthSection = GenericSection<YearMonthSectionType> & {
+  year: number;
+  month: number;
+};
+export type Section = StaticSection | YearMonthSection;
+
+export const groupMessagesByDate = (
+  timestamp: number,
+  messages: Array<Message>
+): Array<Section> => {
+  const referenceDateTime = moment.utc(timestamp);
+  const today = moment(referenceDateTime).startOf('day');
+  const yesterday = moment(referenceDateTime)
+    .subtract(1, 'day')
+    .startOf('day');
+  const thisWeek = moment(referenceDateTime).startOf('isoWeek');
```

Comment:
Is there any chance of time zone problems since we're using an 'iso' option here?

Reply:
According to Wikipedia this is not based on timezones, but it does mean the week always starts on Monday instead of being variable based on locale, e.g. Sunday in the US: https://en.wikipedia.org/wiki/ISO_week_date

This made it a bit more predictable for unit tests, etc.